src/Microsoft.PowerApps.TestEngine/TestEngine.cs

```diff
@@ -36,24 +36,29 @@ public class TestEngine
             _loggerFactory = loggerFactory;
         }
 
-        public async Task<string> RunTestAsync(string testConfigFile, string environmentId, string tenantId, string outputDirectory, string domain, string queryParams)
+        /// <summary>
+        /// This is the entry point function for executing tests. This function expects the Test Plan file along with 
+        /// environment and tenant IDs along with other optional parameters and executes the specified tests on the app
+        /// in the provided tenant and environment.
+        /// </summary>
+        /// <param name="testConfigFile">The (absolute or relative) path to the test plan file.</param>
+        /// <param name="environmentId">The environment ID where the Power App is published.</param>
+        /// <param name="tenantId">The tenant ID where the Power App is published.</param>
+        /// <param name="outputDirectory">The output directory where the test results and logs are to be saved.</param>
+        /// <param name="domain">The domain of the Power Apps Canvas Designer application where the app is published (Example: "apps.powerapps.com").</param>
+        /// <param name="queryParams">Optional query parameters that would be passed to the Player URL for optional features or parameters.</param>
+        /// <returns>The full path where the test results are saved.</returns>
+        /// <exception cref="ArgumentNullException">Throws ArgumentNullException if any of testConfigFile, environmentId, tenantId or domain are missing or empty.</exception>
+        public async Task<string> RunTestAsync(FileInfo testConfigFile, string environmentId, Guid tenantId, DirectoryInfo outputDirectory, string domain, string queryParams)
         {
             // Set up test reporting
             var testRunId = _testReporter.CreateTestRun("Power Fx Test Runner", "User"); // TODO: determine if there are more meaningful values we can put here
             _testReporter.StartTestRun(testRunId);
 
             Logger = _loggerFactory.CreateLogger(testRunId);
 
-            if (string.IsNullOrEmpty(outputDirectory))
```

Comment:
We would still need some null check on this right? if not using the default directory

Reply:
The idea here is to "enforce" a non-null value from the caller, if one isn't provided the caller application won't build even. If we intentionally pass a null object, an exception will be thrown. As for the defaults, we're now delegating that to the caller to decide what the default value should be. In short, it is mandatory to have a value for the arguments when entering into the test engine library. Let me know if this is a reasonable to go with.

Reply:
That's already been taken care of. `SetOutputDirectory` already throws an `ArgumentNullException` that gets caught in this block as well.

Reply:
That is nice. I get that, just an extra safety net what if outputdirectory is null? This will result in null exception.